_posts/reflect-92.md

```diff
@@ -0,0 +1,202 @@
+
+## Spring Boot 
+
+For our post we will be using [Spring Boot](https://spring.io/projects/spring-boot) together with [springdoc-openapi](https://springdoc.org/).
+
+Springdoc integrates nicely with Spring Boot and will help us generate our documentation.
+
+### Example
+
+#### Getting started
+
+To easily get started we only need to add the Springdoc dependency in our `build.gradle`
+
+```groovy
+implementation 'org.springdoc:springdoc-openapi-ui:1.3.3'
+```
+
+First let us define the path of our documentation. This can be defined easily in the application.yml of our project:
+
+```yaml
+springdoc:
+  api-docs:
+    path: /reflectoring-openapi
+```
+
+This, is where our OpenAPI specification lives, which then springdoc uses to beautifully display our endpoints. For more configuration properties please check the [official documentation](https://springdoc.org/springdoc-properties.html).
+
+#### Define APIs Information
+
+Let us define some information about our API.
+
+```java
+@OpenAPIDefinition(
+        info = @Info(
+                title = "Code-First Approach (reflectoring.io)",
+                description = "" +
+                        "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur et rhoncus quam. Aenean quis augue ac eros pulvinar malesuada. " +
+                        "In sagittis elit egestas tincidunt iaculis. " +
+                        "Donec eu lacus vitae nulla varius consectetur a vel quam. Aliquam erat volutpat. Duis eget ullamcorper tellus",
+                contact = @Contact(name = "Reflectoring", url = "https://reflectoring.io", email = "petros.stergioulas94@gmail.com"),
+                license = @License(name = "MIT Licence", url = "https://github.com/thombergs/code-examples/blob/master/LICENSE")),
+        servers = @Server(url = "http://localhost:8080")
+)
+```
+
+Now, start the server and navigate to: http://localhost:8080/swagger-ui/index.html?configUrl=/reflectoring-openapi/swagger-config
+
+You can see now the information that we defined above!
+![General Information](/assets/img/posts/reflect92/general-info.png)
+
+#### Define Rest API
+
+In this section, we will define our Rest endpoint.
+
+```java
+@RequestMapping("/api/todos")
+@Tag(name = "Todo API", description = "euismod in pellentesque massa placerat duis ultricies lacus sed turpis")
+interface TodoApi {
+
+    @GetMapping
+    @ResponseStatus(code = HttpStatus.OK)
+    List<Todo> findAll();
+
+    @GetMapping("/{id}")
+    @ResponseStatus(code = HttpStatus.OK)
+    Todo findById(@PathVariable String id);
+
+    @PostMapping
+    @ResponseStatus(code = HttpStatus.CREATED)
+    Todo save(@RequestBody Todo todo);
+
+    @PutMapping("/{id}")
+    @ResponseStatus(code = HttpStatus.OK)
+    Todo update(@PathVariable String id, @RequestBody Todo todo);
+
+    @DeleteMapping("/{id}")
+    @ResponseStatus(code = HttpStatus.NO_CONTENT)
+    void delete(@PathVariable String id);
+}
+```
+
+The `@Tag` annotation simple describes the endpoint.
```

Comment:
- [ ] Explain where all the annotations come from. Spring? SpringDoc? The reader might be confused.